packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmLogTests.kt

```diff
@@ -196,16 +211,122 @@ class LogTests {
     }
 
     @Test
-    fun equals_same() {
-        val loggers = listOf(createDefaultSystemLogger(Realm.DEFAULT_LOG_TAG))
-        val other = RealmLog(configuration = LogConfiguration(LogLevel.ALL, loggers))
-        assertEquals(log, other)
+    fun addLogger() {
+        val called = atomic<Boolean>(false)
+        val customLogger = object : RealmLogger {
+            override val level: LogLevel = LogLevel.ALL
+            override val tag: String = "CUSTOM"
+            override fun log(
+                level: LogLevel,
+                throwable: Throwable?,
+                message: String?,
+                vararg args: Any?
+            ) {
+                assertEquals("Hello", message)
+                called.value = true
+            }
+        }
+        RealmLog.add(customLogger)
+        RealmLog.trace("Hello")
+        assertTrue(called.value)
+    }
+
+    @Test
+    fun addLogger_twice() {
+        val called = atomic(0)
+        val customLogger = object : RealmLogger {
+            override val level: LogLevel = LogLevel.ALL
+            override val tag: String = "CUSTOM"
+            override fun log(
+                level: LogLevel,
+                throwable: Throwable?,
+                message: String?,
+                vararg args: Any?
+            ) {
+                assertEquals("Hello", message)
+                called.incrementAndGet()
+            }
+        }
+        RealmLog.add(customLogger)
+        RealmLog.add(customLogger)
+        RealmLog.trace("Hello")
+        assertEquals(2, called.value)
+    }
+
+    @Test
+    fun removeLogger_success() {
+        val called = atomic(0)
+        val customLogger = object : RealmLogger {
+            override val level: LogLevel = LogLevel.ALL
+            override val tag: String = "CUSTOM"
+            override fun log(
+                level: LogLevel,
+                throwable: Throwable?,
+                message: String?,
+                vararg args: Any?
+            ) {
+                assertEquals("Hello", message)
+                called.incrementAndGet()
+            }
+        }
+        RealmLog.add(customLogger)
+        assertTrue(RealmLog.remove(customLogger))
+    }
+
+    @Test
+    fun removeLogger_failure() {
```

Comment:
Naming was a bit confusing. Though the `failure` had something to do with `fail()` inside the log method. Maybe rename it?
```suggestion
    fun removeLogger_falseForNonaddedLoggers() {
```